src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationSource.cs

```diff
@@ -31,32 +28,39 @@ public IConfigurationProvider Build(IConfigurationBuilder builder)
             try
             {
                 AzureAppConfigurationOptions options = _optionsProvider();
-                ConfigurationClient client;
+                IConfigurationClientProvider clientProvider;
 
-                if (options.Client != null)
+                if (options.ClientProvider != null)
                 {
-                    client = options.Client;
+                    clientProvider = options.ClientProvider;
                 }
                 else if (!string.IsNullOrWhiteSpace(options.ConnectionString))
                 {
-                    client = _configurationClientFactory.CreateConfigurationClient(options.ConnectionString, options.ClientOptions);
+                    clientProvider = new ConfigurationClientProvider(options.ConnectionString, options.ClientOptions);
                 }
-                else if (options.Endpoint != null && options.Credential != null)
+                else if (options.Endpoints != null && options.Credential != null)
                 {
-                    client = _configurationClientFactory.CreateConfigurationClient(options.Endpoint, options.Credential, options.ClientOptions);
+                    clientProvider = new ConfigurationClientProvider(options.Endpoints, options.Credential, options.ClientOptions);
                 }
                 else
                 {
                     throw new ArgumentException($"Please call {nameof(AzureAppConfigurationOptions)}.{nameof(AzureAppConfigurationOptions.Connect)} to specify how to connect to Azure App Configuration.");
                 }
 
-                provider = new AzureAppConfigurationProvider(client, options, _optional);
+                provider = new AzureAppConfigurationProvider(clientProvider, options, _optional);
             }
-            catch (InvalidOperationException e)
+            catch (InvalidOperationException ex)
             {
                 if (!_optional)
                 {
-                    throw new ArgumentException(e.Message, e);
+                    throw new ArgumentException(ex.Message, ex);
+                }
+            }
+            catch (FormatException fe)
```

Comment:
What's this added for?

Reply:
When the `ConfiguationClientProvider` constructor can't parse the connection_string, it will throw `FormatException`.

Reply:
Is that a new thing? Change in exception propagation is a behavior change.

Reply:
Earlier the parse method would throw `ArgumentException` directly, but now I changed it to `FormatException` (it's an internal method, exception only handled internally).
I had to change because there was one place where we want to suppress exception, but `ArgumentException` was too broad for that use case.

Reply:
Got it, thanks.